src/protocols/device_path.rs

```diff
@@ -52,3 +52,20 @@ impl Hardware {
     pub const SUBTYPE_CONTROLLER: u8 = 0x05;
     pub const SUBTYPE_BMC: u8 = 0x06;
 }
+
+#[repr(C, packed)]
+pub struct HardDriveDevicePathNode {
+    pub header: Protocol,
+    pub partition_number: u32,
+    pub partition_start: u64,
+    pub partition_size: u64,
+    pub partition_signature: [u64; 2],
+    pub partition_format: u8,
+    pub signature_type: u8,
+}
+
+#[repr(C, packed)]
+pub struct HardDriveDevicePath {
+    pub file_system_path_node: HardDriveDevicePathNode,
+    pub end: End,
+}
```

Comment:
I see that this makes creating those structures a lot easier, but I would prefer doing that in a helper crate. This is not really part of the specification, so I would prefer not to include it in `r-efi`.

Reply:
Removed